script/script.go

```diff
@@ -0,0 +1,55 @@
+// Package script provides helpers to interact with scripts.
+package script
+
+//go:generate mockgen -destination=../mocks/mock_runner.go -package=mocks github.com/storageos/init/script Runner
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+)
+
+// Runner is an interface for script runner.
+type Runner interface {
+	// RunScript executes a script at path script, with environment variables
+	// env and arguments arg, returning stdout and stderr byte slices and any
+	// execution error.
+	RunScript(script string, env map[string]string, arg ...string) (stdout []byte, stderr []byte, err error)
+}
+
+// GetAllScripts takes a scripts directory path (absolute path) and scans it for
+// script files, returning a list of all the scripts. It ignores files with docs
+// extensions(.md, .txt).
+func GetAllScripts(scriptsDir string) ([]string, error) {
+	var ignoreFileExt = map[string]bool{
+		".md":  true,
+		".txt": true,
+	}
+
+	allScripts := []string{}
+
+	err := filepath.Walk(scriptsDir, func(path string, info os.FileInfo, err error) error {
```

Comment:
Perhaps we can whitelist rather than blacklist? It's a little more robust if we have to add other artefacts (maybe a Go binary / library / etc) that is called by a script.

If so, we could simplify this to a `filepath.Glob("*.sh")` too

Reply:
We have executable shell scripts and binaries (dp upgrade binary) here, without any identifiable extension. I wasn't aware that we'll have executable binaries as well. Whitelisting based on file extension won't work because scripts are not just shell scripts in this case. Maybe "scripts" is not a good name, "executables" would have been better?
As far as I know, we'll have single file binary/scripts only for now.

Reply:
Ah OK, sorry I thought we were replicating the rc init sort of model where we had a collection of shell scripts that were ran in alphabetic order, shelling out as necessary.

No worries.